src/site/content/en/blog/granular-chunking-nextjs/index.md

```diff
@@ -0,0 +1,273 @@
+splitting. Next.js, for example, would generate a `commons` bundle that contained any module that is
+used in more than 50% of pages and all framework dependencies (`react`, `react-dom`, etc…).
+
+```javascript/11,16
+const splitChunksConfigs = {
+  //...
+  prod: {
+    chunks: 'all',
+    cacheGroups: {
+      default: false,
+      vendors: false,
+      commons: {
+        name: 'commons',
+        chunks: 'all',
+        minChunks: totalPages > 2 ? totalPages * 0.5 : 2,
+      },
+      react: {
+        name: 'commons',
+        chunks: 'all',
+        test: /[\\/]node_modules[\\/](react|react-dom|scheduler|use-subscription)[\\/]/,
+      },
+    },
+  },
+```
+Although including framework-dependent code into a shared chunk means that it can be downloaded and
+cached for any entrypoint, the usage-based heuristic of including common modules used in more than
+_half of pages_ isn't very effective. Modifying this ratio would only result in one of two outcomes:
+more unnecessary code gets downloaded or more code gets duplicated across multiple routes.
+
+To solve this problem, we adopted a [different
+configuration](https://github.com/zeit/next.js/pull/7696) for`SplitChunksPlugin` that reduces
+unecessary code for any route.
+
++   Any sufficiently large third-party module (> 160 KB) is split into its own individual chunk
++   A separate `frameworks` chunk is created for framework dependencies (`react`, `react-dom`,
+    etc…)
++   As many shared chunks as needed are created (up to 25)
+
+This granular chunking strategy provides the following benefits:
+
+1.  **Page load times are improved**. Emitting multiple shared chunks, instead of a single one,
+    minimizes the amount of unneeded (or duplicate) code for any entrypoint.
+2.  **Improved caching during navigations**. Splitting large libraries and framework dependencies
+    into separate chunks reduces the possibility of cache invalidation since both are unlikely to
+    change until an upgrade is made.
+
+## More HTTP requests
+
+`SplitChunksPlugin` defined the basis for granular chunking, and applying this approach to a
+framework like Next.js was not an entirely new concept. Many frameworks, however, still continued to
+use a single heuristic and "commons" bundle strategy for a few reasons. This includes the concern that
+many more HTTP requests can negatively affect site performance.
+
+Browsers can only open a limited number of TCP connections to a single origin (6 for Chrome), so
+minimizing the number of chunks outputted by a bundler can ensure that the total number of requests
+stays under this threshold. However, this only holds true for HTTP/1.1. Multiplexing in HTTP/2
+allows for multiple requests to be streamed in parallel using a single connection over a single
+origin. In other words, we generally do not need to worry about limiting the number of chunks
+emitted by our bundler.
```

Comment:
Should we add an extra note here that multiplexing can also slow down your site as network congestion can happen. Many CDNs don't provide the best experience for this. 

- https://developer.akamai.com/blog/2019/01/31/http2-discover-performance-impacts-effective-prioritization
- https://github.com/andydavies/http2-prioritization-issues
- https://calendar.perfplanet.com/2018/http2-prioritization/

Reply:
Great point, added an aside for this :)